File: Leetcoding-Actions/LC-Monthly-Challenges/2020-11/08-Binary-Tree-Tilt.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int totalTilt = 0;
    
    public int findTilt(TreeNode root) {
        tiltSum(root);
        return totalTilt;
    }
    
    protected int tiltSum(TreeNode root) {
        if (root == null) return 0;
        
        int leftSum = tiltSum(root.left);
        int rightSum = tiltSum(root.right);
        int tilt = Math.abs(leftSum - rightSum);
        totalTilt += tilt;
        
        return leftSum + rightSum + root.val;
    }
}

File: Leetcoding-Actions/LC-Monthly-Challenges/2020-11/08-Binary-Tree-Tilt.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTilt(self, root: TreeNode) -> int:
        
        totalTilt = 0
        def tiltSum(root):
            nonlocal totalTilt
            if not root: return 0
            
            leftSum = tiltSum(root.left)
            rightSum = tiltSum(root.right)
            tilt = abs(leftSum - rightSum)
            totalTilt += tilt
            
            return leftSum + rightSum + root.val
        
        tiltSum(root)
        return totalTilt
